v2/handler.go

```diff
@@ -0,0 +1,110 @@
+package jsonrpc
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+)
+
+// DefaultHandler is the default Handler for serving requests quickly
+var DefaultHandler = &Handler{
+	Dispatcher: DefaultDispatcher,
+}
+
+// Handler provides the interface between http requests and the Dispatcher
+type Handler struct {
+	Dispatcher Dispatcher
+}
+
+func serverError(w http.ResponseWriter, message string, code int) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+
+	result := Response{
+		Version: "2.0",
+		ID:      nil,
+		Error: &Error{
+			Code:    code,
+			Message: message,
+		},
+	}
+
+	resp, _ := Marshal(&result)
+
+	w.Write(resp)
+}
+
+func (handler *Handler) dispatch(resp *Response, call *Call, req *http.Request, wg *sync.WaitGroup) {
+	defer wg.Done()
+	handler.Dispatcher.Dispatch(resp, call, req)
+}
+
+// ServeHTTP handles converting a http.Request into a Calls. Implements the
+// http.Handler interface
+func (handler *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	if r.Method != http.MethodPost {
+		serverError(w, "jsonrpc: rpc calls should be done via a POST request", CodeInvalidRequest)
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		serverError(w, err.Error(), CodeInvalidRequest)
+		return
+	}
+
+	var calls []*Call
+	var single bool
+
+	err = json.Unmarshal(body, &calls)
+
+	if err != nil {
```

Comment:
If call contains a single element doesn't it make sense to send a 1 element array containing that call instead of handling it this way?